askama_derive/src/generator.rs

```diff
@@ -8,11 +8,30 @@ use crate::input::{Source, TemplateInput};
 use crate::CompileError;
 
 use parser::node::{
-    Call, Comment, CondTest, If, Include, Let, Lit, Loop, Match, Target, Whitespace, Ws,
+    Call, Comment, CondTest, FilterBlock, If, Include, Let, Lit, Loop, Match, Target, Whitespace,
+    Ws,
 };
 use parser::{Expr, Node};
 use quote::quote;
 
+struct WriteParts {
+    size_hint: usize,
+    /// First buffer is the "buf format" and second is the "buf expr".
+    ///
+    /// If "buf expr" is None, it means we can generate code like this:
+    ///
+    /// ```ignore
+    /// let var = buf_format;
+    /// ```
+    ///
+    /// Otherwise we need to format the "buf expr" using the "buf format":
+    ///
+    /// ```ignore
+    /// let var = format!(buf_format, buf_expr);
+    /// ```
+    buffers: Option<(Buffer, Option<Buffer>)>,
```

Comment:
Reading this docstring I definitely think we should just make the inner tuple a `struct` with named fields, too.